momaland_demo.ipynb

```diff
@@ -0,0 +1,420 @@
+    "        action = env.action_space(agent).sample()\n",
+    "    env.step(action)\n",
+    "env.close()\n",
+    "\n",
+    "# rewards of all agents from the first step\n",
+    "episode_rewards[0:len(env.possible_agents)]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Step 4.2: Parallel API Demo\n",
+    "The environment is initialized with the `parallel_env()` function. Agents `step()` all at the same time through the environment with their actions. A key difference between AEC and Parallel is that actions and observations are dictionaries in Parallel, as they are all received at the same time."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 66,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'agent_0': array([0., 0., 0.]), 'agent_1': array([0., 0., 0.])}"
+      ]
+     },
+     "execution_count": 66,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from momaland.envs.item_gathering import moitem_gathering_v0\n",
+    "\n",
+    "# new parallel env\n",
+    "env = moitem_gathering_v0.parallel_env()\n",
+    "observations, infos = env.reset()\n",
+    "episode_rewards = []\n",
+    "while env.agents:\n",
+    "    actions = {agent: env.action_space(agent).sample() for agent in env.agents}\n",
+    "    observations, vec_rewards, terminations, truncations, infos = env.step(actions)\n",
+    "    episode_rewards.append(vec_rewards)\n",
+    "env.close()\n",
+    "\n",
+    "# rewards are stored in a dictionary, can be accessed per agent\n",
+    "episode_rewards[0]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Step 5: Wrappers\n",
+    "On top of the native wrappers provided by MOMAland; SuperSuit and PettingZoo wrappers are also compatible with MOMAland environments."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### MOMAland"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 75,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[0.0, 0.0, 0.0]"
+      ]
+     },
+     "execution_count": 75,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from momaland.utils.aec_wrappers import LinearizeReward, NormalizeReward\n",
+    "import numpy as np\n",
+    "\n",
+    "env = momultiwalker_v0.env()\n",
+    "\n",
+    "# Making the vector reward a scalar reward to shift to single-objective multi-agent (aka PettingZoo)\n",
+    "# We can assign different weights to the objectives of each agent.\n",
+    "weights = {\n",
+    "    \"walker_0\": np.array([0.1, 0.7, 0.2]), \n",
+    "    \"walker_1\": np.array([0.6, 0.1, 0.3]), \n",
+    "    \"walker_2\": np.array([0.2, 0.2, 0.6]),\n",
+    "}\n",
+    "env = LinearizeReward(env, weights)\n",
+    "\n",
+    "# Normalizing the reward of each agent\n",
```

Comment:
First normalize, then Linearize

Reply:
<img width="779" alt="image" src="https://github.com/Farama-Foundation/momaland/assets/48724128/f4ec1acb-2cb5-445d-91ec-907e650009a3">
@ffelten we have different weights on the objectives + we normalized it but it doesnt do anything? what am I missing?

Reply:
im appending the wrong variable, i hate asking you questions just to see the answer in 2 seconds even though i looked at it for 10 minutes https://i.imgur.com/lgVG0L2.jpeg